src/modm/driver/radio/sx128x.hpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023, Lasse Alexander Jensen
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SX128X_HPP
+#define MODM_SX128X_HPP
+
+#include <modm/architecture/interface/spi_device.hpp>
+#include <modm/debug/logger.hpp>
+#include <modm/processing/resumable.hpp>
+
+#include "sx128x_definitions.hpp"
+#include "sx128x_hal.hpp"
+
+namespace modm
+{
+
+/**
+ * @tparam	Hal	    Hardware Abstraction Layer for SX128x
+ *
+ * @ingroup modm_driver_sx128x
+ * @author	Lasse Alexander Jensen
+ * @author  Rasmus Kleist Hørlyck Sørensen
+ */
+template< class Hal >
+class Sx128x : public sx128x, public Hal
+{
+public:
+    Sx128x();
+
+    /// Get the transciever status directly.
+    /// @attention this command can be issued at any time as long as it is not the very first command send over the interface
+    modm::ResumableResult<bool>
+    getStatus(Status &status);
+
+    /// Writes a block of bytes in a data memory space starting at a specific address
```

Comment:
I have not fully implemented getStatus yet, because its command structure does not include a wait cycle as all others in SPI. I will return to this when finally having settled on how much responsibility the transport layer should have.